pkg/auth/auth_handler.go

```diff
@@ -163,12 +178,40 @@ func (a *AuthHandler) AuthUserAuth(token string) (interface{}, error) {
 
 	payload, err := parsePayload(parsedToken)
 	if err != nil {
-		a.log.Fatalln("Failed parse payload,", err)
+		a.log.Error("Failed parse payload,", err)
+		return nil, err
+	}
+
+	err = a.storeAdminInPayload(payload)
+	if err != nil {
 		return nil, err
 	}
+
+	if payload.Username == "" {
+		a.log.Error("Missing username in token")
+		return nil, fmt.Errorf("Missing username in token")
+	}
+
 	return payload, nil
 }
 
+func (a *AuthHandler) storeAdminInPayload(payload *ocm.AuthPayload) error {
+	if !a.EnableAuthz {
```

Comment:
If authz is not enabled, I think we still need to store user information in the context (admin user) because the rest of the inventory code relies on it